### ISSUE TITLE
How to get device type like router, laptop or mobile?

### ISSUE THREAD

**Issue by k-sathireddy** [mavris/MMLanScan#63]

**Comment by k-sathireddy** [comment]
I am unable to get the device type using MMLanScan. Can anyone explain me how to acheive this?

**Comment by fukemy** [comment]
any solution?

**Comment by mickeyl** [comment]
I'm not the author, but this is pretty much out-of-scope for this library, which is concerned with finding out whether a device is present. In order to identify net device type, you will have to do further inspection, like checking for open ports and services and then deducting what kind of device it could be.

**Comment by fukemy** [comment]
the biggest problem is getting mac address is blocked now. I dont know why `Fing` can do this on IOS 18

**Comment by mickeyl** [comment]
It doesn't here. Most of the devices in my network stay with the "Generic" label. Those that don't expose services like ZeroConf or SNMP which carries their network address on the protocol level, which is something even Apple can't prevent. (Well, technically they could, but obviously no one told them yet ;-)

**Comment by beeper99** [comment]
> the biggest problem is getting mac address is blocked now. I dont know why `Fing` can do this on IOS 18

It can't, it uses a helper app on a device like a PC to pass the MAC to iOS.
Netsbee does this too with its helper app.https://netsbee.com/helper/
 much better than fing.

**Comment by fukemy** [comment]
please forget the library and the solution, it's impossible now

**Comment by mickeyl** [comment]
Indeed. It's nice to have it here for recording history though. In [NetworkLab](https://apps.apple.com/de/app/network-lab/id6737435505), I use a combination of open services and zeroconf information to guess the device type, but at the end of the day, it's all heuristics. Apple does not want to be the platform for hardcore networking tools, perhaps Android is better in this area.

<img width="603" height="1311" alt="Image" src="https://github.com/user-attachments/assets/fa2c48cc-a10e-415f-ba30-9585cf2a1b5e" />
